periodic/periodic.go

```diff
@@ -497,9 +502,19 @@ func runOne(id int, runnerChan chan struct{},
 	tIDStr := fmt.Sprintf("T%03d", id)
 	perThreadQPS := r.QPS / float64(r.NumThreads)
 	useQPS := (perThreadQPS > 0)
+
 	hasDuration := (r.Duration > 0)
 	useExactly := (r.Exactly > 0)
 	f := r.Runners[id]
+	if useQPS && r.Uniform {
+		delayBetweenRequest := 1. / float64(perThreadQPS)
+		// When using uniform mode, we should wait a bit relative to our QPS and thread ID.
+		// For example, with 10 threads and 1 QPS, thread 8 should delay 0.7s.
+		delaySeconds := delayBetweenRequest - (delayBetweenRequest / float64(r.NumThreads) * float64(r.NumThreads-id))
+		delayDuration := time.Duration(delaySeconds * float64(time.Second))
+		start = start.Add(delayDuration)
+		time.Sleep(delayDuration)
```

Comment:
in theory all sleep times are recorded through
sleepTimes.Record
but maybe this initial offsetting isn't - it will change somewhat the total duration of the run and the end case for the last threads I think (might miss one request at the end?)

what happens if you do
-qps 100 -c 10 -t 10s

in theory each connection would do exactly 100/10*10 = 100 calls and the whole run would be 10s - what does it do with -uniform?

Reply:
Discussed a bit in the above case, it will extend the total duration a bit (100ms here).

I can add `sleepTimes.Record` if you prefer, but it seems like it shouldn't be included

Without it: `count 40 avg 1.2484387 +/- 0.0005214 min 1.2468261840000001 max 1.249850499 sum 49.937547`
With it: `count 50 avg 1.0888235 +/- 0.3443 min 0 max 1.24983677 sum 54.4411737`

We can see it really skews the results

Reply:
thanks for the answers!